src/js/_enqueues/admin/post.js

```diff
@@ -660,7 +660,7 @@ jQuery( function($) {
 			function() {
 				var t = $(this), c = t.is(':checked'), id = t.val();
 				if ( id && t.parents('#taxonomy-'+taxonomy).length ) {
-					$('input[id^="in-' + taxonomy + '-' + id + '"]').prop('checked', c);
+					$('input[id^="in-' + taxonomy + '-' + id + '-"]').prop('checked', c);
```

Comment:
```suggestion
					$('input#in-' + taxonomy + '-' + id + ', input[id^="in-' + taxonomy + '-' + id + '-"]').prop('checked', c);
```

Perhaps this will be better as it would still work on the old HTML. The walker class that outputs that HTML can be changed/replaced by plugins so keeping this bit backwards compatible would be nice.